lib/google/gax/grpc.rb

```diff
@@ -72,22 +74,23 @@ module Grpc
       # @return A gRPC client stub.
       def create_stub(service_path,
                       port,
-                      chan_creds: nil,
                       channel: nil,
+                      chan_creds: nil,
                       updater_proc: nil,
                       scopes: nil)
         address = "#{service_path}:#{port}"
-        if channel.nil?
-          chan_creds = GRPC::Core::ChannelCredentials.new if chan_creds.nil?
+        if channel
+          yield(address, nil, channel_override: channel)
+        elsif chan_creds
```

Comment:
Maybe raise an exception if multiple overriding params are specified instead of silently ignoring some, though.